### PR TITLE
Extended tutorial api

### PR DESCRIPTION
**Pull request by Arcadius19**
Added an extended API for tutorials to:
 - get default tutorial ID: instead of retrieving a tutorial with an ID 0 or 1 as it's been done so far if a requested tutorial does not exist, it should be checked what the actual first tutorial's ID is in case the first one was removed
 - update: there is one function to update an entire tutorial. Should it be split into multiple functions where each tutorial's property is updated (updateTutorialTitle, updateTutorialDescription, updateTutorialSource)?
 - delete

Updates and removals require administration privileges.

### CODE REVIEW

File: api/tutorial.js [hunk 1]
```diff
@@ -1,26 +1,36 @@
 const tutorialDB = require('../db/tutorial-queries')
 
-function createTutorial (req, res, next) {
+// Check if the user is logged-in
+function isLoggedIn (req, res) {
```
Comment (NickWu007) [inline api/tutorial.js]: Extract this into a helper? Other API calls can use this as well.

File: api/tutorial.js [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
       })
-    return
+    return false
   }
+  return true
+}
 
-  // Check if the user has admin rights
+// Check if the user has admin rights
+function isAdmin (req, res) {
```
Comment (NickWu007) [inline api/tutorial.js]: Same. Into a helper.

**Review by NickWu007 — CHANGES_REQUESTED** [review]

**Review by NickWu007 — APPROVED** [review]

**Comment by NickWu007** [comment]
LGTM. Merge when ready.

**Comment by Arcadius19** [comment]
Is there anything else that should be added here now?

**Comment by NickWu007** [comment]
Not that I can think of now, apart from some refactoring work.

We should be able to move to front end after this.

**Pull request merged**